bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java

```diff
@@ -105,11 +101,11 @@ public static BError createError(Type type, BString message, BError cause, BMap<
      * @return new error
      */
     public static BError createError(Type type, BString message, BString details) {
-        MapValueImpl<BString, Object> detailMap =
-                new MapValueImpl<>(setImmutableTypeAndGetEffectiveType(PredefinedTypes.TYPE_ERROR_DETAIL));
+        MappingInitialValueEntry[] initialValues = new MappingInitialValueEntry[1];
```

Comment:
Wouldn't this lead to the same issue where the entry becomes null?

Reply:
In [here](https://github.com/ballerina-platform/ballerina-lang/blob/d042e7043f64f09138932557b2b7be435000542f/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java#L54) for java exceptions like UnsupportedOperationException, the detail message become null.  But when considering the ballerina error values I think detail cannot become null. I kept the null check for detail value as it was in that manner previously

Reply:
Wouldn't the iteration fail in io.ballerina.runtime.internal.values.MapValueImpl#populateInitialValues?

Reply:
It can happen. I have changed this in above mentioned way.